packages/react-dom/src/__tests__/ReactDOMComponent-test.js

```diff
@@ -2283,7 +2286,9 @@ describe('ReactDOMComponent', () => {
 
       expectDev(console.error.calls.count()).toBe(1);
       expectDev(console.error.calls.argsFor(0)[0]).toContain(
-        'Warning: Received `false` for non-boolean attribute `whatever`.',
+        'Received `false` for non-boolean attribute `whatever`. If you mean to conditionally ' +
+          'pass an attribute, use a ternary expression: {`whatever` ? value : undefined} ' +
```

Comment:
Please make sure to read the messages you assert in tests. It looks like you were trying to adjust tests to match the wrong behavior in the code, rather than the other way around.

Reply:
Sorry, my understanding was that the value being passed was the condition. So `value` should hold the attribute name instead of `condition` correct? It didn't make sense that `whatever` (holding the value `false`) could be an attribute.

Reply:
I’m sorry that I snapped. I had a bad day and shouldn’t have spoken to you with this tone. I apologize.

In this test, `whatever` is the *attribute name*. Not the value. You can see if from the first sentence:

>Received `false` for non-boolean attribute `whatever`.

Therefore, a condition like `{`whatever` ? value : undefined}` is not relevant to the user code. The user has something like `whatever={condition && value}` with `condition` being `false`. We want to tell the user to turn it into something like `whatever={condition ? value : undefined}`.

So I think ideally the messages would be:

1. If you get `true`:

```js
expectDev(console.error.calls.argsFor(0)[0]).toContain(
  'Received `true` for non-boolean attribute `whatever`.\n\n' +
  'If you want `true` to appear in the DOM, cast it to a string. ' +
  'For example, you can pass `whatever="true"` or `whatever={String(value)}` instead.'
);
```

2. If you get `false`:

```js
expectDev(console.error.calls.argsFor(0)[0]).toContain(
  'Received `false` for non-boolean attribute `whatever`.\n\n' +
  'If you want `false` to appear in the DOM, cast it to a string. ' +
  'For example, you can pass `whatever="false"` or `whatever={String(value)}`.\n\n' +
  'If you want to omit `whatever` based on a condition, use a ternary expression. ' +
  'For example, you can pass `whatever={condition ? value : undefined}` ' +
  'instead of `whatever={condition && value}`.'
);
```

I hope this is helpful and clarifies things!

Reply:
No need to apologize, your message did not come off badly. I only apologized for making you revise this pr so many times because I did not grasp the whole picture. Yes, this is extremely helpful. Thank you.

Reply:
Do you prefer:
```
`whatever={String(value)}`
```
or
```
`whatever={value.toString()}`
```

Reply:
I thought about the second one but it will throw for null/undefined. Might be unexpected. First one seems safer. Although on the other hand you probably don't want `"undefined"` or `"null"` as strings either.

I wish there was some concise way to express "stringify unless it's null/undefined".

I guess I like that `value.toString()` throws it it doesn't exist. If you mess it up you'll immediately realize it by the crash, instead of quietly using the wrong value. So let's suggest that.